pkg/component/delete/delete.go

```diff
@@ -126,23 +151,19 @@ func (do *DeleteComponentClient) DeleteDevfileComponent(devfileObj parser.Devfil
 		return err
 	}
 
-	// if there are preStop events, execute them before deleting the deployment;
-	// do not fail Delete operation if this fails or if the pod is not running
-	if libdevfile.HasPreStopEvents(devfileObj) {
-		if pod.Status.Phase != corev1.PodRunning {
-			klog.V(4).Infof("unable to execute preStop events, pod for component %q is not running", componentName)
-		} else {
-			klog.V(4).Infof("Executing %q event commands for component %q", libdevfile.PreStop, componentName)
-			// ignore the failures if any; delete should not fail because preStop events failed to execute
-			err = libdevfile.ExecPreStopEvents(devfileObj, componentName, component.NewExecHandler(do.kubeClient, pod.Name, false))
-			if err != nil {
-				klog.V(4).Infof("Failed to execute %q event commands for component %q, cause: %v", libdevfile.PreStop, componentName, err.Error())
-			}
+	// do not fail Delete operation if if the pod is not running or if the event execution fails
+	if pod.Status.Phase != corev1.PodRunning {
+		klog.V(4).Infof("unable to execute preStop events, pod for component %q is not running", componentName)
```

Comment:
To avoid nested blocks, you can return at this point and remove the following `else`